test/e2e/framework/test_context.go

```diff
@@ -530,68 +530,28 @@ func AfterReadingAllFlags(t *TestContextType) {
 	}
 }
 
-const (
-	// This is the traditional gomega.Format default of 4000 for an object
-	// dump plus some extra room for the message.
-	maxFailureMessageSize = 5000
-
-	truncatedMsg = "\n[... see output for full dump ...]\n"
-)
-
 // writeJUnitReport generates a JUnit file in the e2e report directory that is
 // shorter than the one normally written by `ginkgo --junit-report`. This is
 // needed because the full report can become too large for tools like Spyglass
 // (https://github.com/kubernetes/kubernetes/issues/111510).
 //
 // Users who want the full report can use `--junit-report`.
```

Comment:
my understanding is that as long as the `--report-dir` is specified, the junit file will not be a full report.

Reply:
Correct.

Reply:
Let me be more precise: the `junit_01.xml` in `--report-dir` will always be the truncated one. The file written via `--junit-report` will be the complete one.

Reply:
agreed, this is guarded by the checking of `TestContext.ReportDir`.